### DISCUSSION TITLE
trusted publishing with gitlab CI

### DISCUSSION THREAD

**Discussion by HoneyryderChuck** [rubygems/rubygems.org#4845] [Ideas]
Hi, I've had a look at trusted publishing docs, which is a great idea! However, it seems that it only supports github actions at the moment, so I wanted to ask whether you're considering supporting gitlab CI as a trusted partner, as I host most of my gems source code and CI there.

Even if supplying it via UI is far in your roadmap, perhaps self orchestrating it via API is  a possibility? If so, can you give me a quick rundown of the set of api calla I must perform in order to get it done?

Apologies I'd this is not the correct place to ask.

**Comment by simi** [comment]
Hello @HoneyryderChuck, there is PR to add other provider already https://github.com/rubygems/rubygems.org/pull/4159. Feel free submit one for GitLab CI. I'm not sure how that works in there and if we can simply support also custom instances of GitLab for now, but we can discuss details in there.

**Comment by tachyons** [comment]
I've added a [draft PR ](https://github.com/rubygems/rubygems.org/pull/5978)  which adds the support for GitLab CI.
